### PR TITLE
Add new ADR on single-file FB

### PR DESCRIPTION
**Pull request by oerc0122**

### CODE REVIEW

File: documentation/adr/0024-page-to-single-file.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Date: 06-10-2022
+
+## Status
+
+Accepted
+
+## Context
+
+When performing file-backed operations, the [current implementation](0018-algorithm-migration-to-paged-files.md) of
+`PixelData` uses multiple files to store the temporary files and subsequently rebuilds them upon completion of the
```
Comment (cmarooney-stfc) [inline documentation/adr/0024-page-to-single-file.md:15]: Do you mean "rebuilds a single file"

File: documentation/adr/0024-page-to-single-file.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+operation. While this might be efficient for partial (applied to subsections of data) access patterns, this is not the
+case for either full-sequential (applied to whole data [unary/binary ops]) or random-access (applied with variable
```
Comment (cmarooney-stfc) [inline documentation/adr/0024-page-to-single-file.md:17]: For random-access this is fine. For full-sequential, I thought we said that the multiple file storage was fine as each file separately would undergo the operation.

File: documentation/adr/0024-page-to-single-file.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+stride across broad sections of data [cut]). In the Horace code, all operations are performed in either of these two
+modes, as such it is disadvantageous to use multiple temporary files, and instead a single file may be better.
```
Comment (cmarooney-stfc) [inline documentation/adr/0024-page-to-single-file.md:19]: add "and" before "as"

File: documentation/adr/0024-page-to-single-file.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+The reasoning behind a single file being better lies in several factors:
+
+- File system handling of multiple requests for opening can lead to slowdown as system looks up indices. These requests
+  come for every request to open a new file, which will happen multiple times with a series of files.
+- Multiple files may not be sequential on disc leading to inefficient long seek times as multiple processors attempt to
```
Comment (cmarooney-stfc) [inline documentation/adr/0024-page-to-single-file.md:25]: "sequential" - do you mean "contiguous"?

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]
Minor phraseology but worth doing I think

File: documentation/adr/0024-page-to-single-file.md (hunk 4 above)
Comment (oerc0122) [reply documentation/adr/0024-page-to-single-file.md:25]: They might be contiguous but disordered, hence sequential, regardless the contiguity is irrelevant if they are separate files and the disc has to seek to find them in the index-table

**Review by cmarooney-stfc — APPROVED** [review]
with new definition of sequnetial fine approved

**Pull request merged**